module2/homework/insensitivePalindrom/palindrom.cpp

```diff
@@ -0,0 +1,18 @@
+#include "palindrom.hpp"
+
+#include <algorithm>
+#include <cctype>
+
+bool is_palindrome(std::string inputStr) {
+    inputStr.erase(std::remove_if(inputStr.begin(),
+                                  inputStr.end(),
+                                  [](auto strElement) { return !std::isalpha(strElement); }),
+                   inputStr.end());
+
+    std::transform(inputStr.begin(), inputStr.end(), inputStr.begin(),
+                   [](auto strElement) { return std::tolower(strElement); });
+
+    auto pairIterator = std::mismatch(inputStr.begin(), inputStr.end(), inputStr.rbegin());
+
+    return (pairIterator.first == inputStr.end() && pairIterator.second == inputStr.rend());
```

Comment:
You're doing the same job twice here - you could compare just up to half of the string.
Ie. in string
abba

Your code compares
a<->a
b<->b
b<->b
a<->a

Reply:
As I remember we had some example during classes how to do it